sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVectorCore.java

```diff
@@ -197,13 +193,27 @@ public final ColumnVector getChild(int ordinal) {
       StructType st = (StructType) dataType();
       StructField[] fields = st.fields();
       for (int i = 0; i < fields.length; i++) {
-        HostColumnVectorCore tmp = cudfCv.getChildColumnView(i);
-        cachedChildren[i] = new RapidsHostColumnVectorCore(fields[i].dataType(), tmp);
+        cachedChildren[i] = createChildColumnView(i, fields[i].dataType());
       }
     }
     return cachedChildren[ordinal];
   }
 
+  private RapidsHostColumnVectorCore createChildColumnView(int childIndex, DataType dataType) {
+    return createChildColumnView(getBase(), childIndex, dataType);
+  }
+
+  private static RapidsHostColumnVectorCore createChildColumnView(
+      HostColumnVectorCore cv,
+      int childIndex,
+      DataType dataType) {
+    HostColumnVectorCore child = cv.getChildColumnView(childIndex);
+    if (child instanceof HostColumnVector) {
```

Comment:
There are two usages of `RapidsHostColumnVector` in spark-rapids. The first one is unpacking HostColumnVectors in `GpuBroadcastToCpuExec`, which builds nested children as HostColumnVector. The second one is copying to host in `GpuColumnarToRowExec`, which builds nested children as HostColumnVectorCore. HostColumnVector owns a reference counter, but HostColumnVectorCore doesn't.
Perhaps the better approach to fix this problem is to change the cuDF side code: JCudfSerialization.unpackHostColumnVectors. As `copyToHost`, creating `HostColumnVectorCore` for nested children instead of `HostColumnVector`.

Reply:
I filed a cuDF PR to fix it: https://github.com/rapidsai/cudf/pull/9596